File: June-LeetCoding-Challenge/07-Coin-Change-2/Coin-Change-2.java
class Solution {
    public int change(int amount, int[] coins) {
        int[] combinations = new int[amount + 1];
        combinations[0] = 1;
        for (int coin : coins) {
            for (int i = coin; i <= amount; i++) {
                int prev = i - coin;
                if (prev >= 0) {
                    combinations[i] += combinations[prev];
                }
            }
        }
        return combinations[amount];
    }
}

File: June-LeetCoding-Challenge/07-Coin-Change-2/Coin-Change-2.py
class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        if not coins and amount!=0:
            return 0
        dp = [0 for x in range(amount+1)]
        dp[0] = 1
        for coin in coins:
            for i in range(1,amount+1):
                if i >= coin:
                    dp[i] += dp[i-coin]
        return dp[amount]
